packages/walrus/src/client.ts

```diff
@@ -164,12 +180,152 @@ export class WalrusClient {
 			.filter((sliver) => !!sliver);
 
 		if (slivers.length < minSymbols) {
-			throw new Error('Not enough slivers to decode');
+			throw new NotEnoughSliversReceivedError('Not enough slivers to decode');
 		}
 
 		return decodePrimarySlivers(numShards, blobMetadata.metadata.V1.unencoded_length, slivers);
 	}
 
+	/**
+	 * Gets the blob status from multiple storage nodes the returns the latest status that can be verified.
+	 */
+	async getVerifiedBlobStatus({ blobId, signal }: GetVerifiedBlobStatusOptions) {
+		const controller = new AbortController();
+		signal?.addEventListener('abort', () => {
+			controller.abort();
+		});
+
+		// Read from the latest committee because, during epoch change, it is the committee
+		// that will have the most up-to-date information on old and newly certified blobs:
+		const committee = await this.#getActiveCommittee();
+		const stakingState = await this.stakingState();
+		const numShards = stakingState.n_shards;
+
+		const statuses = await new Promise<{ status: BlobStatus; weight: number }[]>(
+			(resolve, reject) => {
+				const results: { status: BlobStatus; weight: number }[] = [];
+				let successWeight = 0;
+				let numNotFoundWeight = 0;
+				let settledCount = 0;
+
+				committee.nodes.forEach((node) => {
+					const weight = node.shardIndices.length;
+
+					this.#storageNodeClient
+						.getBlobStatus({ blobId }, { nodeUrl: node.networkUrl, signal: controller.signal })
+						.then((status) => {
+							if (isQuorum(successWeight, numShards)) {
+								controller.abort('Quorum of blob statuses retrieved successfully.');
+								resolve(results);
+							} else {
+								successWeight += weight;
+								results.push({ status, weight });
+							}
+						})
+						.catch((error) => {
+							if (error instanceof NotFoundError) {
+								numNotFoundWeight += weight;
+							} else if (error instanceof UserAbortError) {
+								reject(error);
+							}
+
+							if (isQuorum(numNotFoundWeight, numShards)) {
+								const abortError = new BlobNotCertifiedError('The blob does not exist.');
+								controller.abort(abortError);
+								reject(abortError);
+							}
+						})
+						.finally(() => {
+							settledCount += 1;
+							if (settledCount === committee.nodes.length) {
+								reject(
+									new NoBlobStatusReceivedError(
+										'Not enough statuses were retrieved to achieve quorum.',
+									),
+								);
+							}
+						});
+				});
+			},
+		);
+
+		const aggregatedStatuses = new Map<string, { status: BlobStatus; totalWeight: number }>();
+		for (const value of statuses) {
+			const { status, weight } = value;
+			const key = await hash(status);
```

Comment:
- Why not just `status.type`?
- If that doesn't work not just `JSON.stringify()`? (this is smaller _end_ result but you need to do this compute anyway and you throw the result on the ground afterwards so this is literally just a worse version of that)